src/libraries/Common/src/System/Numerics/Crc32ReflectedTable.cs

```diff
@@ -0,0 +1,34 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace System.Numerics
+{
+    internal static class Crc32ReflectedTable
+    {
+        internal static uint[] Generate(uint reflectedPolynomial)
+        {
+            uint[] table = new uint[256];
+
+            for (int i = 0; i < 256; i++)
+            {
+                uint val = (uint)i;
+
+                for (int j = 0; j < 8; j++)
```

Comment:
I think this part could be SIMD-vectorized, if supported on CPU. There are CPUs that support SSE and SSE2 but not SSE4.2. For this case this is relevant in my eyes.

## Example intrinsics

### SSE
```h
__m128 _mm_and_ps (__m128 a, __m128 b);
```

### SSE2
```h
__m128i _mm_and_si128 (__m128i a, __m128i b);
```

### AVX
```h
__m256 _mm256_and_ps (__m256 a, __m256 b);
```

### AVX2
```h
__m256i _mm256_and_si256 (__m256i a, __m256i b);
```

Reply:
It's a one-time cost, and a small one at that.  Not worth the added complexity even if it could be vectorized.

Reply:
That it is a one-time cost is true. But vectorizing it would not be a big effort in my opinion, since we have Vector* classes.
Would a vectorization cause any other negative effects code-wise?

Reply:
Depending on what you do, it might block the method from being prejitted.

Reply:
Oh okay then I will leave the vectorization for now. If at some point this method should really lead to a bottleneck, it can still be changed.